File: src/main/java/com/speechpro/empbase/empbase/service/EmployeeService.java
package com.speechpro.empbase.empbase.service;

import com.speechpro.empbase.empbase.model.entities.*;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface EmployeeService {

    List<Employee> getAll();
    List<Employee> getActive();
    Employee getById(Long id);
    Employee getByUname(String uname);
    Employee getByOnesId(String onesId);

    Employee create(Employee employee);

    Employee update(Employee employee);

    void delete(Employee employee);
    void delete(Long id);

    List<Employee> getByDivision(Division division);

    List<Employee> getByLocation(Location location);

    List<Employee> getByPosition(Position position);

    List<Employee> getByOrganization(Organization organization);
}

File: src/main/angular/empbase/src/app/services/employee.service.ts
import { Injectable } from '@angular/core';
import { Employee } from '../models/employee';
import { Http }    from '@angular/http';

@Injectable()
export class EmployeeService{

    constructor(private http: Http){}

    getEmployee(id:number){
        return this.http.get('/api/employee/' + id);
    }

    create(employee:Employee){
        return this.http.post('/api/employee', employee);
    }

    update(employee:Employee){
        return this.http.put('/api/employee', employee);
    }

    delete(id:number){
        if(id) return this.http.delete('/api/employee/' + id);
    }

    getAll(){
        return this.http.get('/api/employee/');
    }

    getActive(){
        return this.http.get('/api/employee/active');
    }

}